packages/backend/src/controllers/runQueryController.ts

```diff
@@ -0,0 +1,177 @@
+import {
+    ApiErrorPayload,
+    ApiQueryResults,
+    MetricQuery,
+} from '@lightdash/common';
+import { Body, Post } from '@tsoa/runtime';
+import express from 'express';
+import {
+    Controller,
+    Middlewares,
+    OperationId,
+    Path,
+    Request,
+    Response,
+    Route,
+    SuccessResponse,
+} from 'tsoa';
+import { projectService } from '../services/services';
+import { allowApiKeyAuthentication, isAuthenticated } from './authentication';
+
+type ApiRunQueryResponse = {
+    status: 'ok';
+    results: {
+        metricQuery: any; // tsoa doesn't support complex types like MetricQuery
+        rows: any[];
+    };
+};
+
+type RunQueryRequest = {
+    // tsoa doesn't support complex types like MetricQuery
+    dimensions: any[]; // Dimensions to group by in the explore
+    metrics: any[]; // Metrics to compute in the explore
+    filters: any;
+    sorts: any[]; // Sorts for the data
+    limit: number; // Max number of rows to return from query
+    tableCalculations: any[]; // calculations to append to results
+    additionalMetrics?: any[]; // existing metric type
+    csvLimit?: number;
+};
+
+@Route('/api/v1/projects/{projectUuid}')
+@Response<ApiErrorPayload>('default', 'Error')
+export class RunViewChartQueryController extends Controller {
+    /**
+     * Run a query for a chart on view mode
+     * @param projectUuid The uuid of the project
+     * @param body chartUuid for the chart to run
+     * @param req express request
+     */
+    @Middlewares([allowApiKeyAuthentication, isAuthenticated])
+    @SuccessResponse('200', 'Success')
+    @Post('/runViewChartQuery')
+    @OperationId('postRunViewChartQuery')
+    async postViewChart(
+        @Body() body: { chartUuid: string },
+        @Path() projectUuid: string,
+        @Request() req: express.Request,
+    ): Promise<ApiRunQueryResponse> {
+        this.setStatus(200);
+        return {
+            status: 'ok',
+            results: await projectService.runViewChartQuery(
+                req.user!,
+                body.chartUuid,
+                projectUuid,
+            ),
+        };
+    }
+
+    /**
+     * Run a query for a chart from dashboards
+     * @param projectUuid The uuid of the project
+     * @param body chartUuid for the chart to run
+     * @param req express request
+     */
+    @Middlewares([allowApiKeyAuthentication, isAuthenticated])
+    @SuccessResponse('200', 'Success')
+    @Post('/runDashboardTileQuery')
```

Comment:
Fixed

Reply:
@rephus this is still on the project controller but it would make more sense on the router/controller for saved charts:

```
POST /api/v1/saved/{savedChartUuid/results
```

In the service for `runViewChartQuery` you have the projectUuid from the `SavedChart` object returned from the saved chart model

Reply:
Done